nvtabular/ops/categorify.py

```diff
@@ -174,6 +174,12 @@ class Categorify(StatOperator):
         value will be `max_size - num_buckets -1`.  Setting the max_size param means that
         freq_threshold should not be given.  If the num_buckets parameter is set,  it must be
         smaller than the max_size value.
+    start_index: int, default 1
```

Comment:
Ben, we could do this, and I personally would be for starting with 0 to be more Pythonic. However from what I understand so far, Categorify() is already starting to output categorical values at 1. For instance, in `nvtabular/test/units/ops.py`, we have 

```
            # again userB has highest frequency given lowest encoding
            assert compare_authors == [2, 5, 1, 3]
            assert compare_engaging == [1, 1, 2, 4]
        else:
            # Column combinations are encoded
            compare_engaging = (
                df_out["Author_Engaging User"].to_list()
                if cpu
                else df_out["Author_Engaging User"].to_arrow().to_pylist()
            )
            assert compare_engaging == [1, 4, 2, 3]
```

So I would suspect existing tests would break with a `start_index` default of 0. What are your thoughts, or perhaps we should do this update as a separate issue to maintain the compatibility when a user does not specify a value of `start_index`?

Reply:
We're starting at 0 - its just that 0 is reserved for out of vocabulary tokens. If we set ```start_index=0``` as the default, everything should work

Reply:
Interesting. Yes, let me take a look and address this point.